problems/[230710] 35. Search Insert Position.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @param {number[]} nums
+ * @param {number} target
+ * @return {number}
+ */
+const searchInsert = function (nums, target) {
+  let left = 0;
+  let right = nums.length - 1;
+
+  while (left < right) {
+    const centerIdx = Math.floor((right + left) / 2);
+
+    if (nums[centerIdx] < target) {
+      left = centerIdx + 1;
+    } else {
+      right = centerIdx;
+    }
+  }
```

Comment:
```suggestion
  while (left < right) {
    const centerIdx = Math.floor((right + left) / 2);

    if (nums[centerIdx] < target) {
      left = centerIdx + 1;
    } else if (nums[centerIdx] > target) {
      right = centerIdx - 1;
    } else if (nums[centerIdx] === target) {
      return centerIdx
    }
  }
```

저는 해당 반복문에서 3가지 경우의 수가 발생할 수 있다고 생각하였어요!
1. `centerIdx`에 해당하는 값이` target`과 동일하다. => `centerIdx`를 리턴
2. `centerIdx`에 해당하는 값이 `target` 보다 작다. => `centerIdx`에서 한칸 움직인 값을 left에 할당
  `centerIdx`보다 `target`이 작으니 `target`은 무조건` centerIdx`의 우항에 존재합니다!
  때문에 `centerIdx + 1`을 `left`에 할당합니다.
3. `centerIdx`에 해당하는 값이 `target` 보다 크다.
  `centerldx`가` target`보다 크니, `target`은 무조건 `centerIdx`의 좌항에 존재합니다!
  때문에 `centerIdx - 1`을 `right`에 할당합니다.